org.eevolution.warehouse/src/main/java/base/org/eevolution/process/GenerateShipmentOutBound.java

```diff
@@ -55,216 +58,214 @@
+
+            if (distributionOrders.get(distributionOrderLine.getDD_Order_ID()) == null)
+                distributionOrders.put(distributionOrderLine.getDD_Order_ID(), distributionOrderLine.getDD_Order());
+
+            distributionOrderLine.setConfirmedQty(outboundLine.getPickedQty());
+            distributionOrderLine.saveEx();
+        }
+
+        // Generate Delivery Manufacturing Order
+        if (outboundLine.getPP_Order_BOMLine_ID() > 0) {
+            MPPOrderBOMLine orderBOMLine = (MPPOrderBOMLine) outboundLine.getPP_Order_BOMLine();
+            if (outboundLine.getPickedQty().subtract(orderBOMLine.getQtyDelivered()).signum() <= 0 && !isIncludeNotAvailable())
+                return;
+
+            MStorage[] storage = MStorage.getAll(getCtx(), orderBOMLine.getM_Product_ID(), outboundLine.getM_LocatorTo_ID(), get_TrxName());
+
+            BigDecimal qtyDelivered = getQtyDelivered(outboundLine, orderBOMLine.getQtyDelivered());
+            List<MPPCostCollector> issues = MPPOrder.createIssue(
+                    orderBOMLine.getParent(),
+                    orderBOMLine,
+                    getMovementDate(),
+                    qtyDelivered,
+                    BigDecimal.ZERO,
+                    BigDecimal.ZERO,
+                    storage,
+                    true);
+
+            issues.forEach(costCollector -> {
+                if (manufacturingIssues.get(costCollector.getPP_Cost_Collector_ID()) == null)
+                    manufacturingIssues.put(costCollector.getPP_Cost_Collector_ID(), costCollector);
+            });
+        }
+    }
+
+    private BigDecimal getQtyDelivered(MWMInOutBoundLine outBoundLine, BigDecimal qtyDemandDelivered) {
+        BigDecimal qtyDelivered;
+
+        if (isIncludeNotAvailable())
+            qtyDelivered = outBoundLine.getQtyToPick().subtract(qtyDemandDelivered);
+        else
+            qtyDelivered = outBoundLine.getPickedQty().subtract(qtyDemandDelivered);
+
+        return qtyDelivered;
+    }
+
+    private void processingIssues() {
+        manufacturingIssues.entrySet().stream().filter(Objects::nonNull)
+                .forEach(entry -> {
+                    MPPCostCollector issue = entry.getValue();
+                    issue.setDocAction(getDocAction());
+                    issue.processIt(getDocAction());
+                    if (!issue.processIt(getDocAction())) {
+                        addLog("@ProcessFailed@ : " + issue.getDocumentInfo());
+                        log.warning("@ProcessFailed@ :" + issue.getDocumentInfo());
+                    }
+                    issue.saveEx();
+                });
+    }
+
+    private void processingShipments() {
+        List<PO> shipmentsToPrint = new ArrayList<PO>();
+        shipments.entrySet().stream().filter(Objects::nonNull).forEach(entry -> {
+            MInOut shipment = entry.getValue();
+            shipment.setDocAction(getDocAction());
+            shipment.processIt(getDocAction());
```

Comment:
Please remove it because already exists a calling for it on next line and document is generated cclosed